tests/unit/Identification/Queue/Mocks/TenantIdentifiableInQueue.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the tenancy/tenancy package.
+ *
+ * Copyright Tenancy for Laravel & Daniël Klabbers <daniel@klabbers.email>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @see https://tenancy.dev
+ * @see https://github.com/tenancy
+ */
+
+namespace Tenancy\Tests\Identification\Queue\Mocks;
+
+use Tenancy\Identification\Contracts\Tenant;
+use Tenancy\Identification\Drivers\Queue\Contracts\IdentifiesByQueue;
+use Tenancy\Identification\Drivers\Queue\Events\Processing;
+use Tenancy\Testing\Mocks\Tenant as Mock;
+
+class TenantIdentifiableInQueue extends Mock implements IdentifiesByQueue
+{
+    /**
+     * Specify whether the tenant model is matching the queue job.
+     *
+     * @param Processing $event
+     *
+     * @return Tenant
+     */
+    public function tenantIdentificationByQueue(Processing $event = null): ?Tenant
+    {
+        return $this->newQuery()->first();
+    }
```

Comment:
Let's put some actual logic here for testing. Simply returning first is not a real use case.
Check the HTTP Identification Driver Tests, you didn't simply return first there either.